typescript/sdk/src/token/adapters/SealevelTokenAdapter.ts

```diff
@@ -516,6 +516,14 @@ export class SealevelHypNativeAdapter extends SealevelHypTokenAdapter {
   }
 
   override async getBalance(owner: Address): Promise<bigint> {
+    console.log({ owner });
+    if (eqAddress(owner, this.addresses.warpRouter)) {
+      const collateralAccount = this.deriveNativeTokenCollateralAccount();
+      console.log({ collateralAccount });
+      const balance = await this.getProvider().getBalance(collateralAccount);
```

Comment:
any reason to not do `this.wrappedNative.getBalance(collateralAccount)`? Just to play well with the existing abstractions

Reply:
I initially tried this but its a pubkey vs an address type